crates/iota-names-indexer/src/metrics.rs

```diff
@@ -26,11 +26,37 @@ impl IotaNamesMetrics {
     }
 }
 
-pub(crate) static METRICS: OnceLock<Arc<IotaNamesMetrics>> = OnceLock::new();
+pub(crate) struct PrometheusServer {
+    registry_service: RegistryService,
+}
+
+impl PrometheusServer {
+    pub(crate) fn new() -> Self {
+        Self {
+            registry_service: RegistryService::new(Registry::new()),
+        }
+    }
+
+    pub(crate) async fn start(&self, token: CancellationToken) -> anyhow::Result<()> {
+        let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), 9184);
+        info!("Starting prometheus metrics at {}", addr);
 
-pub(crate) fn start_prometheus_server() -> Registry {
-    let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), 9184);
-    info!("Starting prometheus metrics at http://0.0.0.0:9184/metrics");
+        let app = Router::new()
+            .route(METRICS_ROUTE, get(iota_metrics::metrics))
+            .layer(Extension(self.registry_service.clone()));
```

Comment:
Why do we need to go through axum now?

Reply:
It already was but this way we can control the graceful shutdown of the server